framework/src/base/MooseApp.C

```diff
@@ -1206,6 +1206,28 @@ MooseApp::registerRestartableData(const std::string & name,
   return *insert_pair.first->second.value;
 }
 
+bool
+MooseApp::hasRestartableData(const std::string & name,
```

Comment:
Rather than have these error checks, can you just create a hasRestartableMetaData method?

Reply:
That sounds cleaner. The only issue is that this function will be different from `registerRestartableData`, which put meta data and normal data together. I slightly prefer two separate functions. If you still want this change. I can certainly do.